File: frontend/src/types/task.js
/**
 * Task object shape
 * @typedef {Object} Task
 * @property {string} id
 * @property {string} title
 * @property {string} description
 * @property {string} deadline
 * @property {string} priority - 'high' | 'medium' | 'low'
 * @property {string} status - 'pending' | 'in_progress' | 'completed'
 * @property {string} source - 'email' | 'manual' | 'ai_generated'
 */

export const TaskSample = {
  id: '',
  title: '',
  description: '',
  deadline: '',
  priority: 'medium',
  status: 'pending',
  source: 'manual',
}

File: backend/app/schemas/task.py
from pydantic import BaseModel
from typing import Optional


class TaskCreate(BaseModel):
    title: str
    description: Optional[str] = ""
    deadline: Optional[str] = None
    priority: str = "medium"
    source: str = "manual"


class TaskResponse(BaseModel):
    id: str
    title: str
    description: str
    deadline: Optional[str] = None
    priority: str = "medium"
    status: str = "pending"
    source: str = "manual"


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    deadline: Optional[str] = None
    priority: Optional[str] = None
    status: Optional[str] = None
